### DISCUSSION TITLE
Sending raw PCM bytes over WebSocket doesn't work with Bun (same code as Node), need help

### DISCUSSION THREAD

**Discussion by kynnyhsap** [deepgram/community#740] [General help]
### Which Deepgram product feature are you using?

Deepgram API - STT Streaming

### Details

(IMPORTANT: this is NOT related to the issues of running `@deepgram/sdk` with bun, I use raw webosckets)

I wrote a simple webscoket server that reads audio file (raw PCM bytes) and sends chunks over to deepgram. Works fine with `node`, but when executed in `bun` i find that deepgram does not send me any messages back as if it is ignoring the data i sent. 

I'll attach my code that works similarly both in bun and node v22 (they both support browser-like webosckets api via uWebScoket, but i also tried with popular `ws` npm package and result was identical).

I used bun with deepgram before for other audio formats and it worked ok, i can't figure out whats different with PCM this time. i recorded that PCM file on my raspberry pi, but again, it works fine with node.

I tried to setup a weboscket server to emulate deepgram and checked if data sent with bun vs. node any different - but this experiment showed no difference between the data sent from node or bun.

My only hope is for you guys to look at the request ids and tell if there is any difference in the data being sent.

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

Here are request ids for the node and bun:
* node
  * id `136727d0-baf1-4d47-ad06-b7482841344c` 
  * url `POST /v1/listen?model=nova-2&language=en-US&channels=2&sample_rate=44100&encoding=linear16&extra=runtime%3Anode`
* bun
  * id `3293aa5d-c873-4a3a-a09a-a7ef96dea075`
  * url `POST /v1/listen?model=nova-2&language=en-US&channels=2&sample_rate=44100&encoding=linear16&extra=runtime%3Abun`

one is resolved and one is pending:

<img width="1873" alt="Screenshot 2024-05-04 at 14 30 33" src="https://github.com/deepgram/community/assets/30318777/df6dd004-bb02-4cb4-8244-4582816894f4">

### If possible, please attach your code or paste it into the text box.

My webscoket server code

to run with node v22: `node index.js`
to run with bun: `bun run index.js`

index.js:
```js
import fs from "fs";

const DEEPGRAM_API_KEY = "";

const params = {
  model: "nova-2",
  language: "en-US",
  channels: 2,
  sample_rate: 44100,
  encoding: "linear16",
  extra: `time:${Date.now()}`,
};

console.time("latency");
const ws = new WebSocket(
  `https://api.deepgram.com/v1/listen?${new URLSearchParams(params).toString()}`,
  {
    headers: { Authorization: `Token ${DEEPGRAM_API_KEY}` },
  },
);

ws.addEventListener("open", async () => {
  console.timeEnd("latency");
  console.log("connection opened");

  ws.addEventListener("error", (e) => console.error("connection error", e));
  ws.addEventListener("close", (e) =>
    console.log("connection closed:", e.reason),
  );

  ws.addEventListener("message", (e) => {
    console.log("message", e.data);
  });

  const fileStream = fs.createReadStream("audio.pcm");

  for await (const chunk of fileStream) {
    console.log("sending chunk of size", chunk.length);
    ws.send(chunk);
  }
});

```

### If possible, please attach an example audio file to reproduce the issue.

Couldn't upload it right here but here is the link to my PCM file (12 sec long) via google drive https://drive.google.com/file/d/1an81JdNR3oGhSwefKdHANFeub_knKWfo/view?usp=sharing

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment]
Hey @kynnyhsap, another user ran into issues with Bun as well. It's not clear if Bun does not thoroughly support websockets, or if there is an issue with the code. The other user swapped to Node.

I posted in the Bun forums [here](https://github.com/oven-sh/bun/discussions/8462) with the information I had.

I can see the duration of the audio for those request IDs, and the audio duration for the Node request ID is 11.71 seconds, and the duration for the Bun request is 0. No data was sent when using Bun.

I know Bun uses a special interface for working with websockets, and I haven't closely inspected your code to see if there are important differences. Here are Bun's docs: https://bun.sh/docs/api/websockets
